internal/configs/test_file.go

```diff
@@ -537,6 +654,181 @@ func decodeTestRunOptionsBlock(block *hcl.Block) (*TestRunOptions, hcl.Diagnosti
 	return &opts, diags
 }
 
+func decodeOverrideResourceBlock(block *hcl.Block, mode addrs.ResourceMode) (*OverrideResource, hcl.Diagnostics) {
```

Comment:
It looks like there's some duplication between the next few functions that could be refactored out?

Reply:
Yep, there's an almost identical function that could be moved out. Thanks!

[8b7f372](https://github.com/opentofu/opentofu/pull/1499/commits/8b7f372415f9a4dccbfad433212f9a6a5fbaa1bf)